src/app/pages/questions/components/question/timed-test/timed-test.component.ts

```diff
@@ -66,54 +71,51 @@ export class TimedTestComponent implements OnInit, OnChanges {
       this.timer.start = 0
     }
 
-    this.taskTimer = <TaskTimer>{
-      seconds: this.timer.start,
-      runTimer: false,
+    this.taskTimer = {
       hasStarted: false,
       hasFinished: false,
-      secondsRemaining: this.timer.start
+      secondsElapsed: 0,
+      secondsRemaining: this.timer.start,
+      duration: this.timer.start - this.timer.end,
+      displayTime: this.timer.start
     }
-
-    this.taskTimer.displayTime = this.taskTimer.secondsRemaining.toString()
   }
 
   startTimer() {
     this.taskTimer.hasStarted = true
-    this.taskTimer.runTimer = true
+    this.startTime = Date.now()
+    this.endTime =
+      this.startTime + getMilliseconds({ seconds: this.taskTimer.duration })
     this.timerTick()
   }
 
-  pauseTimer() {
-    this.taskTimer.runTimer = false
-  }
-
-  resumeTimer() {
-    this.startTimer()
+  updateCountdown() {
+    this.taskTimer.secondsElapsed = Math.floor(
+      getSeconds({
+        milliseconds: Date.now() - this.startTime
+      })
+    )
+    this.taskTimer.displayTime =
+      this.timer.start - this.taskTimer.secondsElapsed
   }
 
   timerTick() {
-    setTimeout(() => {
-      if (!this.taskTimer.runTimer) {
-        return
-      }
-      if (this.taskTimer.secondsRemaining > 0) {
-        this.taskTimer.secondsRemaining--
-        this.taskTimer.displayTime = this.taskTimer.secondsRemaining.toString()
-        if (this.taskTimer.secondsRemaining > this.timer.end) {
-          this.timerTick()
-        } else {
-          this.dialogs.beep(1)
-          this.vibration.vibrate(600)
+    if (!this.taskTimer.hasStarted) {
+      return
+    }
+    this.updateCountdown()
+    if (this.endTime - Date.now() > 0)
+      setTimeout(() => {
+        this.timerTick()
+      }, 1000)
+    else this.stopTimer()
```

Comment:
You can use `setInterval` here like - 
```suggestion
     let timerId = setInterval(() => {
        this.updateCountdown()
        
        if (this.endTime - Date.now() < 0) {
           clearInterval(timerId)
           this.stopTimer()
        }
      }, 1000)
```

This should be sufficient unless recursion is required. Also remember to clear the interval or timeout otherwise it will not be collected by the Garbage Collector